crates/interfaces/src/blockchain_tree/error.rs

```diff
@@ -167,7 +170,12 @@ impl InsertBlockErrorKind {
 
     /// Returns true if this is a block pre merge error.
     pub fn is_block_pre_merge(&self) -> bool {
-        matches!(self, InsertBlockErrorKind::Execution(BlockExecutionError::BlockPreMerge { .. }))
+        matches!(
+            self,
+            InsertBlockErrorKind::Execution(BlockExecutionError::Validation(
+                BlockValidationError::BlockPreMerge { .. }
```

Comment:
it seems like we have some debt around massively nesting error types

Reply:
yeah, I created this https://github.com/paradigmxyz/reth/issues/2903, but it might be stale now that the error types are nested, ultimately we should have a validation error type that we can return either from stages, or from the tree. this error would be used:
 * in the pipeline to unwind + yield the failing block so the engine can invalidate the block
 * in the tree, also yielding the failing block so the caller can invalidate